internal/service/alarms/internal/db/migrations/000005_create_alarm_subscription_info_table.up.sql

```diff
@@ -1,15 +1,22 @@
 -- Stores information about alarm subscriptions
+-- Ref: O-RAN.WG6.O-CLOUD-IM.0-R004-v02.00
 CREATE TABLE IF NOT EXISTS alarm_subscription_info (
     -- O-RAN
     subscription_id UUID PRIMARY KEY DEFAULT gen_random_uuid(), -- Unique identifier for each subscription
-    consumer_subscription_id UUID, -- Optional ID for the consumer's subscription identifier
-    filter TEXT, -- Criteria for filtering alarms, if any
+    -- consumerSubscriptionID does not explicitly set UUID in format (likely someone forgot to add it), however, inventory subscription does, so we will use UUID
+    consumer_subscription_id UUID NULL, -- Optional ID for the consumer's subscription identifier
+    -- filter set nullable as false, but description says that if not set, all alarms are included
+    -- setting as NULL based on description and model definition in O-RAN.WG6.O2IMS-INTERFACE-R004-v07.00
+    filter VARCHAR(20) NULL, -- Can be [new, change, clear, acknowledge], NULL means all
     callback TEXT NOT NULL, -- URL or endpoint for sending notifications
 
     -- Internal
     event_cursor BIGINT NOT NULL DEFAULT 0, -- Tracks the latest event for the subscriber. This used with alarm_event_record.alarm_sequence_number
     created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP, -- Record creation timestamp
     updated_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP -- Record last update timestamp
+
+    -- check does not fail with NULL
+    CONSTRAINT chk_filter CHECK (filter IN ('new', 'change', 'clear', 'acknowledge'))
```

Comment:
It would be best if this check was done in our rest handler so that we can provide a properly formatted error message rather than rely on the message returned by the DB.

Reply:
rest handler will also have a check